functional-tests/src/test/java/org/infinispan/online/service/endpoint/HotRodTester.java

```diff
@@ -36,7 +37,8 @@ public HotRodTester(String serviceName, OpenShiftClient client) {
    public void clear(URL urlToService) {
       RemoteCacheManager cachingService = getRemoteCacheManager(urlToService, true);
       RemoteCache<String, String> defaultCache = cachingService.getCache();
-      defaultCache.clear();
+      CloseableIteratorSet<String> entriesToBeDeleted = defaultCache.keySet();
+      entriesToBeDeleted.stream().forEach(k -> defaultCache.remove(k));
```

Comment:
I'm not against this, but I don't understand why it's necessary?

Reply:
@wburns mentioned that invoking `clear` causes allocating off-heap objects back on-heap. And this is what I'm seeing in the test. Most of them die when invoking `clear`.

Reply:
Ok sounds good. Thanks Will!

Reply:
I created https://github.com/infinispan/jdg/pull/1510